src/mongo/db/query/datetime/date_time_support.h

```diff
@@ -292,8 +286,12 @@ class TimeZone {
     void insertPadded(OutputStream& os, int number, int width) const {
         invariant(width >= 1);
         invariant(width <= 4);
-        invariant(number >= 0);
-        invariant(number <= 9999);
+
+        uassert(18537,
+                str::stream() << "$dateToString is only defined on date components 0-9999,"
```

Comment:
While you're here, I believe there are plans to use this code for other things than $dateToString at some point (started but not committed as part of [SERVER=29627](https://jira.mongodb.org/browse/SERVER-29627)) Can you update this to instead be `"Could not convert date to string, date component was outside supported range of 0-9999: " << number` or something like that?

Reply:
Done!